easybuild/toolchains/compiler/craypewrappers.py

```diff
@@ -0,0 +1,182 @@
+Support for the Cray Programming Environment Wrappers (aka cc, CC, ftn).
+The Cray compiler wrappers are actually way more than just a compiler drivers.
+
+The basic concept is that the compiler driver knows how to invoke the true underlying
+compiler with the compiler's specific options tuned to Cray systems.
+
+That means that certain defaults are set that are specific to Cray's computers.
+
+The compiler wrappers are quite similar to EB toolchains as they include
+linker and compiler directives to use the Cray libraries for their MPI (and network drivers)
+Cray's LibSci (BLAS/LAPACK et al), FFT library, etc.
+
+
+@author: Petar Forai (IMP/IMBA, Austria)
+@author: Kenneth Hoste (Ghent University)
+"""
+
+from easybuild.tools.config import build_option
+from easybuild.tools.toolchain.compiler import Compiler
+from easybuild.toolchains.compiler.gcc import Gcc
+from easybuild.toolchains.compiler.inteliccifort import IntelIccIfort
+
+
+TC_CONSTANT_CRAYPEWRAPPER = "CRAYPEWRAPPER"
+
+
+class CrayPEWrapper(Compiler):
+    """Generic support for using Cray compiler wrappers"""
+
+    # no toolchain components, so no modules to list here (empty toolchain definition w.r.t. components)
+    # the PrgEnv and craype are loaded, but are not considered actual toolchain components
+    COMPILER_MODULE_NAME = []
+    COMPILER_FAMILY = TC_CONSTANT_CRAYPEWRAPPER
+
+    COMPILER_UNIQUE_OPTS = {
+        # FIXME: (kehoste) how is this different from the existing 'shared' toolchain option? just map 'shared' to '-dynamic'? (already done)
+        'dynamic': (True, "Generate dynamically linked executables and libraries."),
+        'mpich-mt': (False, "Directs the driver to link in an alternate version of the Cray-MPICH library which \
+                             provides fine-grained multi-threading support to applications that perform \
+                             MPI operations within threaded regions."),
+        'usewrappedcompiler': (False, "Use the embedded compiler instead of the wrapper"),
+    }
+
+    COMPILER_UNIQUE_OPTION_MAP = {
+        'pic': 'shared',
+        'shared': 'dynamic',
+        'static': 'static',
+        'verbose': 'craype-verbose',
+        'mpich-mt': 'craympich-mt',
+    }
+
+    COMPILER_CC = 'cc'
+    COMPILER_CXX = 'CC'
+
+    COMPILER_F77 = 'ftn'
+    COMPILER_F90 = 'ftn'
+
+    # FIXME (kehoste) hmmmm, really? then how do you control optimisation, precision when using the Cray wrappers?
+    COMPILER_FLAGS = []  # we dont have this for the wrappers
+    COMPILER_OPT_FLAGS = []  # or those
+    COMPILER_PREC_FLAGS = []  # and those for sure not !
+
+    # template and name suffix for PrgEnv module that matches this toolchain
+    # e.g. 'gnu' => 'PrgEnv-gnu/<version>'
+    PRGENV_MODULE_NAME_TEMPLATE = 'PrgEnv-%(suffix)s/%(version)s'
+    PRGENV_MODULE_NAME_SUFFIX = None
+
+    # template for craype module (determines code generator backend of Cray compiler wrappers)
+    CRAYPE_MODULE_NAME_TEMPLATE = 'craype-%(optarch)s'
```

Comment:
also name,suffix and version (the fact that the `name` comes from `optarch` option, doesn't require that you use it like that in the template)